### ISSUE TITLE
Create ndarray backed by CUDAarray UnownedPointer from c++

### ISSUE THREAD

**Issue by starmaid** [cupy/cupy#8348]
### Description

At a high level, I'm trying to pass a CUDA TextureObject from c++ into an ndarray in CuPy, where I can then do operations with it on the GPU.

Because TextureObjects are backed by a CUDAarray, the logical method seems to be get a pointer to the CUDAarray that backs the data of the texture and pass it to CuPy (I can pass other attributes separately). However, the `cupy.cuda.texture.CUDAarray().ptr` attribute is read-only and allocated by the constructor. 

Trying to use a CUDAarray pointer (via `UnownedMemory()` and `MemoryPointer()`) as the `memptr` for a `cupy.ndarray` causes a `cudaErrorIllegalAddress` on an attempted access, because I think its treating it as a linear memory pointer. 

Clearly ndarray backed by texture memory is already a feature from #2432 , but I don't think the API is ready for passing pointers the same way that linear memory allocated with cudaMalloc is.

### Additional Information

I also have a [stackoverflow question](https://stackoverflow.com/questions/78572493/access-cudaarray-in-cupy-using-pointer-from-c) that goes into example code for my application.

**Comment by leofang** [comment]
As you already alluded to, there's no way to address a CUDA Array in a linear fashion, i.e. pointer arithmetics does not work. So I don't think it is reasonable to use it to back an ndarray, which requires the underlying memory to be addressable via pointers (with proper offsets).

The TODO comment that you noticed 
https://github.com/cupy/cupy/blob/d8df1f0405bb645c84274551a7630cf0247d6dd5/cupy/cuda/texture.pyx#L287-L288
was left by me when I was learning about how things work 😅 If you find it confusing perhaps we can remove it. Feel free to send a PR!

**Comment by leofang** [comment]
Forgot to ask the most important question: What are you trying to achieve with CUDA Arrays / texture objects? Perhaps I can show you a path if you share more. Your SO question did not mention anything related to that, only that you want to use it to back an ndarray (for what purpose?)

**Comment by starmaid** [comment]
> which requires the underlying memory to be addressable via pointers

That answers the question! Sorry for getting distracted by the comment, I might PR to remove it. I've totally been there!

My goal is to create a tensor in pytorch (possibly using `torch.from_numpy()`?) from the CUDAarray, without the data leaving the GPU. Someone has a working example of creating a [tensor from an ndarray using CuPy](https://discourse.panda3d.org/t/allowing-torch-tensorflow-to-directly-access-rendered-image-in-gpu/29119/9), at least.

My CUDAarray is coming from a `cudaGraphicsResource` I get from `cudaGraphicsD3D11RegisterResource`, with `cudaGraphicsSubResourceGetMappedArray` to get the CUDAarray reference. Turns out I don't need a whole TextureObject as I stated at the beginning of the question.
